src/content/blog/please-stop-sending-me-nested-dependency-security-reports/index.mdx

```diff
@@ -0,0 +1,97 @@
+pubDate: 2024-04-29
+title: "Please Stop Sending Me Nested Dependency Security Reports"
+---
+
+import LabeledVideo from "~/components/blog/mdx/LabeledVideo.astro";
+
+As an open source maintainer, I regularly receive "critical security vulnerability" DMs, emails, and GitHib issues about my projects.
+Most of these notices are sent from individual developers or IT teams about packages that are dependencies of dependencies.
+These "nested dependency" reports often go something like:
+
+<blockquote>
+Hello.
+We use `your-package` and `npm audit` is reporting a critical issue:
+<br />
+```plaintext
+some-nested-dependency  <=0.1.2
+Severity: critical
+Prototype Pollution in some-nested-dependency - https://github.com/advisories/GHSA-ab12-cd23-ef45
+No fix available
+node_modules/some-dependency/node_modules/some-nested-dependency
+  some-dependency  >=2.3.4
+  Depends on vulnerable versions of some-nested-dependency
+  node_modules/some-dependency
+    your-package  >=5.6.7
+    Depends on vulnerable versions of some-dependency
+    node_modules/your-package
+```
+<br />
+Could you please fix this?
+</blockquote>
+
+I appreciate the attention to detail and security.
+Really, I do!
+Paying attention to your tooling is a useful skill for any developer.
+
+But these reports aren't useful.
+They don't come with the context and proof-of-concept a real vulnerability report needs to.
+I've never seen one indicate a real vulnerability.
+
+Reproduction-less reports about nested dependencies' security reports _spam maintainers_ like me and actually _hamper security_ by training us to ignore reports.
+Unless you can demonstrate a real security vulnerability in one of my packages, please don't send me a report about a nested dependency.
+
+<LabeledVideo
+	alt="Larry David in Curb Your Enthusiasm looking more and more consternated, subtly talking to himself."
+	description="Captured: an open source maintainer trying to find phrasing to discourage these reports while still encouraging security as a practice"
+	original="https://giphy.com/gifs/MyzfERo9SqG1G"
+	src="/images/blog/larry-david-consternation.webm"
+/>
+
+## This Is Probably Not a Real Vulnerability
+
+These excessive reports are produced because scanners such as npm audit and Dependabot don't have a way to indicate which API(s) in a package are impacted by a reported issue.
+That means every downstream package that relies on any package -even if they never use the impacted APIs- will get a security report.
+
+Most of the time, the vulnerability is only exploitable if a user runs some carefully crafted malicious code on the server the package is running on.
+That can be very important to patch for some users of some packages!
+
+But: many packages are only ever used at development time.
+A linter plugin, for example, isn't likely vulnerable to attacks that require modifying JavaScript global objects or providing exponentially expensive regular expressions to function calls.
```

Comment:
Genuinely curious, why not? I'd imagine a linter has write permissions and can modify global objects and perform regexes? If that's not the case, might be worth explaining why linters can't do those things.

Reply:
Settled on:

> A linter plugin, for example, will often only be run on safely parsed representations of code written by the project's developers.
> That's not a realistic vector for attacks that require passing a raw untrusted strings to specific API.